File: lost.lua
local Lost = {}

function Lost:new()
    self.text = love.graphics.newText(love.graphics.newFont(40), "You lost")
    self.textX = (love.graphics.getWidth() - self.text:getWidth()) / 2
    self.textY = (love.graphics.getHeight() - self.text:getHeight()) / 2
end

function Lost:keypressed(key)
    if key == "escape" then
        Application.popState()
    end
end 

function Lost:draw()
    love.graphics.draw(self.text, self.textX, self.textY)
end

local Lost__mt = {
    __index = Lost
}

return setmetatable(Lost, {
    __call = function(_, ...)
        local o = setmetatable({}, Lost__mt)
        o:new(...)
        return o
    end
})

File: src/lost.ts
import { KeyConstant } from "love.keyboard";
import { GameState } from "./gamestate";

export class Lost extends GameState {
    private text: Text;
    private textPosition: Vector;

    constructor() {
        super();
        this.text = love.graphics.newText(love.graphics.newFont(40), "You lost");
        this.textPosition = {
            x: (love.graphics.getWidth() - this.text.getWidth()) / 2,
            y: (love.graphics.getHeight() - this.text.getHeight()) / 2,
        };
    }

    getName() {
        return "Lost";
    }

    keypressed(key: KeyConstant) {
        if (key == "escape") {
            Application.popState();
        }
    }

    draw() {
        love.graphics.draw(this.text, this.textPosition.x, this.textPosition.y);
    }
}
